### ISSUE TITLE
add acl option to s3_bucket_object

### ISSUE THREAD

**Issue by gozer** [hashicorp/terraform#7951]
I've been hit by this limitation today, using s3_bucket_object to create a file in an S3 bucket works as expected.

However, that object will get the default *private* ACLs applied to it by S3, and there is no way to change that.

I'd love if there was a way to specify the acl in creating the object with the canned ACLs, like 'public-read', 'private', 'bucket-owner-read', etc.

```tf
resource "aws_s3_bucket_object" "hello" {
    bucket = "test-bucket"
    key = "index.html"
    content = "Hello world from Terraform!"
    content_type = "text/plain"
    #acl = "public-read"
}
```

**Comment by kwilczynski** [comment]
I wonder if exposing storage class would be helpful too?

**Comment by gozer** [comment]
Certainly would be, as would supporting whatever else the underlying API does.

**Comment by kwilczynski** [comment]
Related to #7744.

**Comment by stack72** [comment]
Closed via #8091
